_publications/jors/paper.tex

```diff
@@ -0,0 +1,362 @@
+frontend processing to eliminate the need for content management
+systems~\cite{cmsfree}, which are generally costly to maintain
+especially for small scientific communities with limited funding and
+staffing.
+
+The workflow for uploading benchmark results relies on third party
+tools using the following steps, illustrated in
+Figure~\ref{fig:pfhub_website}.
+
+\begin{itemize}
+  \item The users are first required to upload simulation outputs to
+    an archival resource (\emph{e.g.}, Figshare~\footnote{Certain
+      commercial equipment, instruments, or materials (or suppliers,
+      or software, ...) are identified in this paper to foster
+      understanding. Such identification does not imply recommendation
+      or endorsement by the National Institute of Standards and
+      Technology, nor does it imply that the materials or equipment
+      identified are necessarily the best available for the
+      purpose.\label{disclaimer}}) configured with permissive
+    cross-origin resource sharing (CORS).
+  \item The metadata summarizing each simulation is entered into a
+    form on the website, including relevant details such as memory
+    usage, run time and links to the data archived in the first step.
+  \item Upon submission, the Staticman app~\cite{staticman} submits
+    the entered metadata as a GitHub pull request to the PFHub GitHub
+    repository.  The metadata is stored in a YAML file with a unique
+    path in the repository.
+  \item Travis CI~\cite{travis} performs linting on the submission and
+    then launches a temporary version of the proposed website using
+    Surge~\cite{surge}. The PFHub admins can then examine the new
+    submission and further changes can be made if necessary.
+  \item Once review has been completed to the satisfaction of both the
+    uploading scientist and the website maintainers, the pull request
+    is merged and served to the World Wide Web using a hosting service
+    compatible with GitHub Pages.
+\end{itemize}
+
+A combination of Jekyll templates and Coffeescript are used to access
+and download the data links in the submitted YAML files and then
+display the data in interactive plots on the website. The interactive
+plots are displayed using the Plotly JavaScript Graphing
+Library~\cite{plotly} as it provides a programmable interface and
+requires minimal configuration.
+
+The benchmark specifications consist of equations, narrative, plots
+and code samples, and are composed in Jupyter Notebooks. The Jupyter
+Notebooks are included as static objects in the website after
+translation into HTML using the nbconvert tool~\cite{jupyter}.
+
+The combination of a central repository on GitHub for website source
+code and metadata with distributed data records on third-party
+archives avoids the complexity and administrative overhead of
+maintaining a live database and associated back-end application. The
+PFHub infrastructure provides a template for other small scientific
```

Comment:
"The PFHub infrastructure provides a template for other small scientific communities to host custom content and integrate data from members of their community."

This is the thesis statement.  It should go at the front of the section.

Then, I would state that the general applicability of the PFHub framework will be described in the specific instance of the phase field benchmarking effort.  As such, other communities would likely want to change the exact details of what is included (for example, they might not want benchmark problems in Jupyter notebooks but rather something else) but the general approach and even specific code framework is transferable.

Reply:
I've rewritten the start of the Implementation and Architecture section and have the "The PFHub infrastructure provides ..." sentence at the beginning. Also an extra sentence about how the framework is transferable.